target/linux/ar71xx/image/tp-link.mk

```diff
@@ -669,6 +669,22 @@ define Device/tl-wr1041n-v2
 endef
 TARGET_DEVICES += tl-wr1041n-v2
 
+define Device/tl-wr1043n-v5
+  $(Device/tplink)
+  DEVICE_TITLE := TP-LINK TL-WR1043N v5
+  BOARDNAME := TL-WR1043N-v5
+  DEVICE_PROFILE := TLWR1043
+  TPLINK_HWID :=  0x10430005
+  MTDPARTS := spi0.0:128k(factory-uboot)ro,128k(u-boot)ro,1536k(kernel),13568k(rootfs),128k(product-info)ro,640k(config)ro,64k(partition-table)ro,128k(logs)ro,64k(ART)ro,15104k@0x40000(firmware)
```

Comment:
You can merge kernel and rootfs partition to firmware partition.
Other partition used in TP-Link firmware You can merge to one tp-link partition. Look to MTDPARTS on tl-wr902ac device.